.github/mingw-build.sh

```diff
@@ -0,0 +1,174 @@
+    CFLAGS="-mthreads -mtune=generic -O2 -pipe"
+    CXXFLAGS="${CFLAGS}"
+    LDFLAGS="-pipe -static-libgcc -static-libstdc++"
+    export CYANRIPBUILDDIR CYANRIPINSTALLDIR PKG_CONFIG_PATH CPATH LIBRARY_PATH CPPFLAGS CFLAGS CXXFLAGS LDFLAGS
+    mkdir -p "$CYANRIPBUILDDIR" "$CYANRIPINSTALLDIR" || true
+
+    cyan_do_hide_all_sharedlibs
+    sed -ri 's;-R[^ ]*;;g' "$MINGW_PREFIX/lib/pkgconfig/libcdio.pc"
+}
+
+cyan_do_vcs() {
+    local vcsURL=${1#*::} vcsFolder=$2
+    local vcsBranch=${vcsURL#*#} ref=origin/HEAD
+    : "${vcsFolder:=$(basename "$vcsURL" .git)}"
+
+    cd $CYANRIPBUILDDIR
+    git clone "$vcsURL" "$vcsFolder"
+    cd "$vcsFolder"
+}
+
+cyan_create_build_dir () {
+    [[ -d _build ]] && rm -rf ./_build
+    mkdir _build
+    cd _build
+}
+
+cyan_do_hide_all_sharedlibs() {
+    local files
+    files="$(find /mingw{32,64}/lib /mingw{32/i686,64/x86_64}-w64-mingw32/lib -name "*.dll.a" 2> /dev/null)"
+    local tomove=()
+    for file in $files; do
+        [[ -f "${file%*.dll.a}.a" ]] && tomove+=("$file")
+    done
+    printf '%s\n' "${tomove[@]}" | xargs -ri mv -f '{}' '{}.dyn'
+}
+
+cyan_hide_files() {
+    local reverse=n echo_cmd
+    [[ $1 == "-R" ]] && reverse=y && shift
+    [[ $dryrun == y ]] && echo_cmd="echo"
+    for opt; do
+        if [[ $reverse == n ]]; then
+            [[ -f $opt ]] && $echo_cmd mv -f "$opt" "$opt.bak"
+        else
+            [[ -f "$opt.bak" ]] && $echo_cmd mv -f "$opt.bak" "$opt"
+        fi
+    done
+}
+cyan_hide_conflicting_libs() {
+    # meant for rude build systems
+    local reverse=n
+    [[ $1 == "-R" ]] && reverse=y && shift
+    local priority_prefix
+    local -a installed
+    mapfile -t installed < <(find "$CYANRIPINSTALLDIR/lib" -maxdepth 1 -name "*.a")
+    if [[ $reverse == n ]]; then
+        hide_files "${installed[@]//$CYANRIPINSTALLDIR/$MINGW_PREFIX}"
+    else
+        hide_files -R "${installed[@]//$CYANRIPINSTALLDIR/$MINGW_PREFIX}"
+    fi
+    if [[ -n $1 ]]; then
+        priority_prefix="$1"
+        mapfile -t installed < <(find "$priority_prefix/lib" -maxdepth 1 -name "*.a")
+        if [[ $reverse == n ]]; then
+            hide_files "${installed[@]//$1/$CYANRIPINSTALLDIR}"
+        else
+            hide_files -R "${installed[@]//$1/$CYANRIPINSTALLDIR}"
+        fi
+    fi
+}
+
+cyan_do_separate_confmakeinstall() {
+    cyan_create_build_dir
+    ../configure --disable-shared --enable-static --prefix="$CYANRIPINSTALLDIR" "$@" &&
+    make &&
+    make install
+}
+
+cyan_do_cmake() {
+    cyan_create_build_dir
+    PKG_CONFIG=pkg-config cmake .. -G Ninja -DBUILD_SHARED_LIBS=off \
+        -DCMAKE_INSTALL_PREFIX="$CYANRIPINSTALLDIR" -DUNIX=on \
+        -DCMAKE_FIND_ROOT_PATH="$(cygpath -m "$CYANRIPINSTALLDIR") $(cygpath -m "$MINGW_PREFIX") $(cygpath -m "$MINGW_PREFIX/$MINGW_CHOST")" \
+        -DCMAKE_PREFIX_PATH="$(cygpath -m "$CYANRIPINSTALLDIR") $(cygpath -m "$MINGW_PREFIX") $(cygpath -m "$MINGW_PREFIX/$MINGW_CHOST")" \
```

Comment:
```suggestion
        -DCMAKE_FIND_ROOT_PATH="$(cygpath -pm "$CYANRIPINSTALLDIR:$MINGW_PREFIX:$MINGW_PREFIX/$MINGW_CHOST")" \
        -DCMAKE_PREFIX_PATH="$(cygpath -pm "$CYANRIPINSTALLDIR:$MINGW_PREFIX:$MINGW_PREFIX/$MINGW_CHOST")" \
```

`CMAKE_FIND_ROOT_PATH` and `CMAKE_PREFIX_PATH` is supposed to be `;` separated